kvbc/src/ReplicaImp.cpp

```diff
@@ -89,103 +91,89 @@ Status ReplicaImp::stop() {
+                          std::vector<std::optional<categorization::Value>> &values) const {
+  return m_kvBlockchain->multiGet(category_id, keys, versions, values);
 }
 
-Status ReplicaImp::addBlock(const SetOfKeyValuePairs &updates,
-                            BlockId &outBlockId,
-                            const concordUtils::SpanWrapper & /*parent_span*/) {
-  // TODO(GG): check legality of operation (the method should be invoked from
-  // the replica's internal thread)
+void ReplicaImp::multiGetLatest(const std::string &category_id,
+                                const std::vector<std::string> &keys,
+                                std::vector<std::optional<categorization::Value>> &values) const {
+  return m_kvBlockchain->multiGetLatest(category_id, keys, values);
+}
 
-  // TODO(GG): what do we want to do with several identical keys in the same
-  // block?
+std::optional<categorization::TaggedVersion> ReplicaImp::getLatestVersion(const std::string &category_id,
+                                                                          const std::string &key) const {
+  return m_kvBlockchain->getLatestVersion(category_id, key);
+}
 
-  TimeRecorder scoped_timer(*histograms_.add_block);
-  return addBlockInternal(updates, outBlockId);
+void ReplicaImp::multiGetLatestVersion(const std::string &category_id,
+                                       const std::vector<std::string> &keys,
+                                       std::vector<std::optional<categorization::TaggedVersion>> &versions) const {
+  return m_kvBlockchain->multiGetLatestVersion(category_id, keys, versions);
 }
 
-void ReplicaImp::deleteGenesisBlock() {
-  const auto genesisBlock = m_bcDbAdapter->getGenesisBlockId();
-  if (genesisBlock == 0) {
-    throw std::logic_error{"Cannot delete the genesis block from an empty blockchain"};
-  }
-  m_bcDbAdapter->deleteBlock(genesisBlock);
+std::optional<categorization::Updates> ReplicaImp::getBlockUpdates(BlockId block_id) const {
+  return m_kvBlockchain->getBlockUpdates(block_id);
 }
 
-BlockId ReplicaImp::deleteBlocksUntil(BlockId until) {
-  const auto genesisBlock = m_bcDbAdapter->getGenesisBlockId();
-  if (genesisBlock == 0) {
-    throw std::logic_error{"Cannot delete a block range from an empty blockchain"};
-  } else if (until <= genesisBlock) {
-    throw std::invalid_argument{"Invalid 'until' value passed to deleteBlocksUntil()"};
-  }
+BlockId ReplicaImp::getGenesisBlockId() const { return m_kvBlockchain->getGenesisBlockId(); }
 
-  const auto lastBlock = getLastBlock();
-  const auto lastDeletedBlock = std::min(lastBlock, until - 1);
-  for (auto i = genesisBlock; i <= lastDeletedBlock; ++i) {
-    m_bcDbAdapter->deleteBlock(i);
+BlockId ReplicaImp::getLastBlockId() const {
+  const auto lastSt = m_kvBlockchain->getLastStatetransferBlockId();
+  if (lastSt) {
+    return *lastSt;
   }
-  return lastDeletedBlock;
+  return m_kvBlockchain->getLastReachableBlockId();
```

Comment:
As discussed offline, I think we should return the last reachable, we can do it after we merge